spec/controllers/account/tasks_controller_spec.rb

```diff
@@ -3,20 +3,35 @@
 RSpec.describe Account::TasksController, type: :controller do
   render_views
 
-  context 'PUT /:move' do
-    let!(:user) { create(:user) }
-    let!(:workspace) { create(:workspace, user_id: user.id) }
-    let!(:project) { create(:project, workspace_id: workspace.id) }
-    let!(:task) { create_list(:task, 5, project_id: project.id) }
-
-    before do
-      sign_in user
-      user.update(projects: [project])
+  let!(:user) { create(:user) }
+  let!(:workspace) { create(:workspace, user: user) }
+  let!(:project) { create(:project, workspace: workspace) }
+  let!(:task) { create_list(:task, 5, project: project) }
+
+  before do
+    sign_in user
+    user.update(projects: [project])
```

Comment:
fyi
```
let!(:user) { create(:user, projects: [project] } 
```

Reply:
why u have such complicated assoc chain?
shouldn't it be like `user :has_many projects thtough: :workplace`?